src/main/java/de/unistuttgart/iste/gits/content_service/service/ContentService.java

```diff
@@ -0,0 +1,88 @@
+package de.unistuttgart.iste.gits.content_service.service;
+
+import de.unistuttgart.iste.gits.generated.dto.ContentDto;
+import de.unistuttgart.iste.gits.generated.dto.CreateContentInputDto;
+import de.unistuttgart.iste.gits.generated.dto.UpdateContentInputDto;
+import de.unistuttgart.iste.gits.content_service.persistence.dao.ContentEntity;
+import de.unistuttgart.iste.gits.content_service.persistence.mapper.ContentMapper;
+import de.unistuttgart.iste.gits.content_service.persistence.repository.ContentRepository;
+import de.unistuttgart.iste.gits.content_service.validation.ContentValidator;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
```

Comment:
I don't have this annotation is other services. Why is it needed? If only needed in a specific method, consider only having it there. #Nitpick

Reply:
Transactional is needed when a service is accessing different entities, e.g. tag and content. Otherwise you get some obscure hibernate exception. In the content microservice, we plan to offer an API where tags are a list of string attributes for a content object, hence the ContentService class will have to deal with content and tag entities. For simplicity reasons, I would like to keep the transactional annotation on class level.